File: bindings/lua/Polycode/BoneTrack.lua
class "BoneTrack"


function BoneTrack:__getvar(name)
	if name == "position" then
		local retVal = Polycode.BoneTrack_get_position(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "scale" then
		local retVal = Polycode.BoneTrack_get_scale(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "boneQuat" then
		local retVal = Polycode.BoneTrack_get_boneQuat(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Quaternion"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "weight" then
		return Polycode.BoneTrack_get_weight(self.__ptr)
	end
end

function BoneTrack:__setvar(name,value)
	if name == "position" then
		Polycode.BoneTrack_set_position(self.__ptr, value.__ptr)
		return true
	elseif name == "scale" then
		Polycode.BoneTrack_set_scale(self.__ptr, value.__ptr)
		return true
	elseif name == "boneQuat" then
		Polycode.BoneTrack_set_boneQuat(self.__ptr, value.__ptr)
		return true
	elseif name == "weight" then
		Polycode.BoneTrack_set_weight(self.__ptr, value)
		return true
	end
	return false
end
function BoneTrack:BoneTrack(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.BoneTrack(unpack(arg))
	end
end

function BoneTrack:Play(once)
	local retVal = Polycode.BoneTrack_Play(self.__ptr, once)
end

function BoneTrack:Stop()
	local retVal =  Polycode.BoneTrack_Stop(self.__ptr)
end

function BoneTrack:Update(elapsed)
	local retVal = Polycode.BoneTrack_Update(self.__ptr, elapsed)
end

function BoneTrack:Reset()
	local retVal =  Polycode.BoneTrack_Reset(self.__ptr)
end

function BoneTrack:setSpeed(speed)
	local retVal = Polycode.BoneTrack_setSpeed(self.__ptr, speed)
end

function BoneTrack:__delete()
	if self then Polycode.delete_BoneTrack(self.__ptr) end
end

File: bindings/javascript/Polycode/BoneTrack.js
function BoneTrack(bone,length) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.BoneTrack(bone,length)
	}
	Object.defineProperties(this, {
		'position': { enumerable: true, configurable: true, get: BoneTrack.prototype.__get_position, set: BoneTrack.prototype.__set_position},
		'scale': { enumerable: true, configurable: true, get: BoneTrack.prototype.__get_scale, set: BoneTrack.prototype.__set_scale},
		'boneQuat': { enumerable: true, configurable: true, get: BoneTrack.prototype.__get_boneQuat, set: BoneTrack.prototype.__set_boneQuat},
		'weight': { enumerable: true, configurable: true, get: BoneTrack.prototype.__get_weight, set: BoneTrack.prototype.__set_weight}
	})
}


BoneTrack.prototype.__get_position = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.BoneTrack__get_position(this.__ptr)
	return retVal
}

BoneTrack.prototype.__set_position = function(val) {
	Polycode.BoneTrack__set_position(this.__ptr, val.__ptr)
}

BoneTrack.prototype.__get_scale = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.BoneTrack__get_scale(this.__ptr)
	return retVal
}

BoneTrack.prototype.__set_scale = function(val) {
	Polycode.BoneTrack__set_scale(this.__ptr, val.__ptr)
}

BoneTrack.prototype.__get_boneQuat = function() {
	var retVal = new Quaternion("__skip_ptr__")
	retVal.__ptr = 	Polycode.BoneTrack__get_boneQuat(this.__ptr)
	return retVal
}

BoneTrack.prototype.__set_boneQuat = function(val) {
	Polycode.BoneTrack__set_boneQuat(this.__ptr, val.__ptr)
}

BoneTrack.prototype.__get_weight = function() {
	return Polycode.BoneTrack__get_weight(this.__ptr)
}

BoneTrack.prototype.__set_weight = function(val) {
	Polycode.BoneTrack__set_weight(this.__ptr, val)
}

Duktape.fin(BoneTrack.prototype, function (x) {
	if (x === BoneTrack.prototype) {
		return;
	}
	Polycode.BoneTrack__delete(x.__ptr)
})

BoneTrack.prototype.Play = function(once) {
	Polycode.BoneTrack_Play(this.__ptr, once)
}

BoneTrack.prototype.Stop = function() {
	Polycode.BoneTrack_Stop(this.__ptr)
}

BoneTrack.prototype.Update = function(elapsed) {
	Polycode.BoneTrack_Update(this.__ptr, elapsed)
}

BoneTrack.prototype.Reset = function() {
	Polycode.BoneTrack_Reset(this.__ptr)
}

BoneTrack.prototype.setSpeed = function(speed) {
	Polycode.BoneTrack_setSpeed(this.__ptr, speed)
}
